packages/common/src/types/filter.ts

```diff
@@ -199,6 +199,60 @@ export const updateFieldIdInFilters = (
     }
 };
 
+export const isMetricToDelete = (
+    item: FilterGroupItem,
+    metricName: string,
+): boolean => !isFilterGroup(item) && item.target.fieldId === metricName;
+
+export const removeMetricFromFilterGroupItem = (
+    item: FilterGroupItem,
+    metricName: string,
+): void => {
+    if (isFilterGroup(item)) {
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        removeMetricFromFilters(item, metricName);
+    }
+};
+
+export const removeMetricFromFilters = (
```

Comment:
Can we add some unit tests? 

should return remaining filters ( simple case + nested groups )
should return undefined when all filter rules where removed ( simple case + nested groups )